core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java

```diff
@@ -658,6 +669,57 @@ public void updateUsage(StorageTierAssoc globalStorageTierAssoc, List<String> st
     mUsage.updateUsage(globalStorageTierAssoc, storageTiers, totalBytesOnTiers, usedBytesOnTiers);
   }
 
+  /**
+   * Require the lock for update replica info.
+   */
+  LockResource lockReplicaInfoBlock() {
+    return new LockResource(mReplicaInfoLock);
+  }
+
+  /**
+   * @param blockId the id of the block whose replica number is changed
+   * @param AddedNum the changed value
+   * Update the replica changed map.
+   */
+  public void updateReplica(long blockId, long AddedNum) {
+    try (LockResource r = lockReplicaInfoBlock()) {
+      if(mRequiredSyncReplica || mReplicaNum.size() > 10000){
+        mRequiredSyncReplica = true;
+        mReplicaNum.clear();
+      }
+      else if (mReplicaNum.containsKey(blockId)) {
+        mReplicaNum.put(blockId, (short)(mReplicaNum.get(blockId) + (short)AddedNum));
+      } else {
+        mReplicaNum.put(blockId, (short)AddedNum);
+      }
+    }
+  }
+
+  /**
+   * @return the changed replica info to be sent to workers
+   */
+  public Map<Long, Long> getReplicaInfo(BlockMetaStore mBlockMetaStore) {
+    Map<Long, Long> retReplicaNum = new HashMap<>();
```

Comment:
Why `long` as the value type, while `mReplicaNum` uses `short`?

Reply:
Proto cannot transport short type. And the memory pressure is limited for proto.